File: Tests/Tests/STPPaymentMethodSEPADebitTest.m
//
//  STPPaymentMethodSEPADebitTest.m
//  StripeiOS Tests
//
//  Created by Cameron Sabol on 10/7/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "STPTestUtils.h"

@interface STPPaymentMethodSEPADebitTest : XCTestCase

@end

@implementation STPPaymentMethodSEPADebitTest

#pragma mark - STPAPIResponseDecodable Tests

- (void)testDecodedObjectFromAPIResponseRequiredFields {
    NSArray<NSString *> *requiredFields = @[];

    for (NSString *field in requiredFields) {
        NSMutableDictionary *response = [[STPTestUtils jsonNamed:@"SEPADebitSource"][@"sepa_debit"] mutableCopy];
        [response removeObjectForKey:field];

        XCTAssertNil([STPPaymentMethodSEPADebit decodedObjectFromAPIResponse:response]);
    }

    XCTAssert([STPPaymentMethodSEPADebit decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"SEPADebitSource"][@"sepa_debit"]]);
}

- (void)testDecodedObjectFromAPIResponseMapping {
    NSDictionary *response = [STPTestUtils jsonNamed:@"SEPADebitSource"][@"sepa_debit"];
    STPPaymentMethodSEPADebit *sepaDebit = [STPPaymentMethodSEPADebit decodedObjectFromAPIResponse:response];

    XCTAssertEqualObjects(sepaDebit.bankCode, @"37040044");
    XCTAssertEqualObjects(sepaDebit.branchCode, @"a_branch");
    XCTAssertEqualObjects(sepaDebit.country, @"DE");
    XCTAssertEqualObjects(sepaDebit.fingerprint, @"NxdSyRegc9PsMkWy");
    XCTAssertEqualObjects(sepaDebit.last4, @"3001");
    XCTAssertEqualObjects(sepaDebit.mandate, @"NXDSYREGC9PSMKWY");

    XCTAssertNotEqual(sepaDebit.allResponseFields, response);
    XCTAssertEqualObjects(sepaDebit.allResponseFields, response);
}


@end

File: Stripe/StripeiOSTests/STPPaymentMethodSEPADebitTest.swift
//  Converted to Swift 5.8.1 by Swiftify v5.8.28463 - https://swiftify.com/
//
//  STPPaymentMethodSEPADebitTest.m
//  StripeiOS Tests
//
//  Created by Cameron Sabol on 10/7/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

class STPPaymentMethodSEPADebitTest: XCTestCase {
    // MARK: - STPAPIResponseDecodable Tests

    func testDecodedObjectFromAPIResponseRequiredFields() {
        let requiredFields: [String]? = []

        for field in requiredFields ?? [] {
            var response = STPTestUtils.jsonNamed("SEPADebitSource")["sepa_debit"] as? [AnyHashable: Any]
            response?.removeValue(forKey: field)

            XCTAssertNil(STPPaymentMethodSEPADebit.decodedObject(fromAPIResponse: response))
        }

        XCTAssertNotNil(STPPaymentMethodSEPADebit.decodedObject(fromAPIResponse: STPTestUtils.jsonNamed("SEPADebitSource")["sepa_debit"] as? [AnyHashable: Any]))
    }

    func testDecodedObjectFromAPIResponseMapping() {
        let response = STPTestUtils.jsonNamed("SEPADebitSource")["sepa_debit"] as? [AnyHashable: Any]
        let sepaDebit = STPPaymentMethodSEPADebit.decodedObject(fromAPIResponse: response)

        XCTAssertEqual(sepaDebit?.bankCode, "37040044")
        XCTAssertEqual(sepaDebit?.branchCode, "a_branch")
        XCTAssertEqual(sepaDebit?.country, "DE")
        XCTAssertEqual(sepaDebit?.fingerprint, "NxdSyRegc9PsMkWy")
        XCTAssertEqual(sepaDebit?.last4, "3001")
        XCTAssertEqual(sepaDebit?.mandate, "NXDSYREGC9PSMKWY")

        XCTAssertEqual(sepaDebit!.allResponseFields as NSDictionary, response! as NSDictionary)
    }
}
